### ISSUE TITLE
Custom field (formula) and custom expression forms are broken

### ISSUE THREAD

**Issue by luxemate** [metabase/metabase#4906]
It's not possible to move cursor in Field Formula input as it instantly moves back to start position. It also doesn't offer me to choose an operation (+,- etc.), unlike Chrome in Windows does. In Chrome it's not even possible to blur from Field Formula input.

![image](https://cloud.githubusercontent.com/assets/1465003/25521320/c8b2c106-2c07-11e7-9b18-369ccbabaeb1.png)

Warnings and errors in Chrome console:
![image](https://cloud.githubusercontent.com/assets/1465003/25521325/cd86e770-2c07-11e7-982c-16899c014661.png)

- Your browser and the version: Chrome 58.0, Firefox 53.0
- Your operating system: Ubuntu Linux 16.04.2 LTS
- Your databases: Postgres
- Metabase version: 0.23.1
- Metabase hosting environment: JAR
- Metabase internal database: H2 (default)

**Comment by LukeAbell** [comment]
Same issue here.

**Comment by mazameli** [comment]
Oh yikes, yeah. Here I've tried to write the word `runs` but the cursor moves back to the start after each key press, rendering the word backwards:

![screen shot 2017-04-28 at 10 31 04 am](https://cloud.githubusercontent.com/assets/2223916/25539867/e1df7c5a-2bfd-11e7-9a10-93d309af35d2.png)

**Comment by mazameli** [comment]
Motion to add this to 0.24. It seems like a feature-breaking bug.

**Comment by VikramTiwari** [comment]
@mazameli I would argue that this is a hotfix that should go out as 0.23.2. Even if it's not a hotfix issue for the project, it's a bug that everyone will run into as their browsers get updated and a working release is breaking.

**Comment by kdoh** [comment]
So after a bit of digging, it seems that in Chrome / FF the cursor position is [always getting set](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/dom.js#L100) to `0, 0` on every key press which is why things end up getting rendered backwards and the cursor can't ever leave the initial position.

I haven't been able to figure out exactly why that's happening specifically in those browsers and not in Safari (and therefore the mac app) yet though.

**Comment by tlrobinson** [comment]
I was able to reduce it down to this: https://jsfiddle.net/tlrobinson/wryfspsq/

When you have an empty input and enter a character, in Safari you get:

```
onSelectionChange 1
onSelectionChange 1
onSelectionChange 1
```

Whereas in Chrome 58 you get:

```
onSelectionChange 1
onSelectionChange 1
onSelectionChange 0
onSelectionChange 0
onSelectionChange 0
onSelectionChange 0
```

Haven't figured out anything beyond this yet.

**Comment by salsakran** [comment]
is this a bug in Chrome/FF or just a new behaviour? i.e. if we wait it out, will it get fixed on its own?

**Comment by tlrobinson** [comment]
Unclear because I'm not sure what the root cause is yet, but given it's happening in both FF and Chrome it seems like it's probably something we should fix.

**Comment by tlrobinson** [comment]
Also do we know if it ever worked on Firefox?

**Comment by VikramTiwari** [comment]
Not working on Firefox 49.0 not even on 53.0 (64-bit, Latest).

**Comment by tlrobinson** [comment]
@luxemate @LukeAbell @VikramTiwari I've got a fix for this in #4969, if you'd like to try it out.
